pkg/k8s/watchers/watcher.go

```diff
@@ -572,15 +572,19 @@ func genCartesianProduct(
 	}
 
 	svcs := make([]loadbalancer.SVC, 0, svcSize)
+	feFamilyIPv6 := utils.IsIPv6(fe)
 
 	for fePortName, fePort := range ports {
 		var besValues []loadbalancer.Backend
 		for ip, backend := range bes.Backends {
-			if backendPort := backend.Ports[string(fePortName)]; backendPort != nil {
+			parsedIP := net.ParseIP(ip)
```

Comment:
some hackity-hack: `isv6 := strings.Contains(ip, ":")` (but leave the `parsedIP := net.ParseIP(ip)` since we will use it :-) )

Reply:
I was thinking of making the use of `IsIPv6` consistent, so its easy to understand simply looking at the function what it informs.